code/modules/antagonists/_common/antag_datum.dm

```diff
@@ -76,7 +76,7 @@ GLOBAL_LIST_EMPTY(antagonists)
 /datum/antagonist/proc/is_banned(mob/M)
 	if(!M)
 		return FALSE
-	. = (jobban_isbanned(M, ROLE_SYNDICATE) || QDELETED(M) || (job_rank && (jobban_isbanned(M,job_rank) || QDELETED(M))))
+	. = (is_banned_from(M.client.ckey, list(ROLE_SYNDICATE, job_rank)) || QDELETED(M))
```

Comment:
Client can and most likely is null here, same in few other places. Just mob.ckey should be fine.